identifyyeastcells.py

```diff
@@ -0,0 +1,1146 @@
+        """
+
+        result = self.get_object_measurements(pipeline, object_name, category,
+                                              {self.object_name.value: []})
+        if category == C_OBJECT_FEATURES:
+            result += [FTR_OBJECT_QUALITY]
+        return result
+
+    ui_to_precision_map = dict([(1, 9), (2, 11), (3, 12), (4, 13), (5, 15)])
+    precision_to_ui_map = dict([(v, k) for k, v in ui_to_precision_map.items()] + [(10, 2)])
+
+    @property
+    def decoded_segmentation_precision_value(self):
+        return self.ui_to_precision_map[self.segmentation_precision.value]
+
+    @memory_profile
+    def run(self, workspace):
+        input_image_name = self.input_image_name.value
+        self.current_workspace = workspace
+        image_set = workspace.image_set
+
+        # same image_set for fitting
+        self.fitting_image_set = image_set
+
+        #
+        # Load images from workspace
+        #
+        input_pixels, background_pixels, ignore_mask_pixels = self.get_input_images_from_image_set(image_set)
+
+        input_image = image_set.get_image(input_image_name, must_be_grayscale=True)
+        self.input_image_file_name = input_image.file_name
+        self.current_workspace.display_data.input_pixels = input_pixels
+
+        #
+        # Preprocessing
+        #
+        input_pixels, background_pixels, ignore_mask_pixels = self.preprocess_images(input_pixels, background_pixels, ignore_mask_pixels)
+
+        #
+        # Segmentation
+        #
+        objects, objects_qualities, background_pixels = self.segmentation(input_pixels, background_pixels, ignore_mask_pixels)
+        objects.parent_image = input_image
+
+        if self.__get(F_BACKGROUND, workspace, None) is None and self.background_elimination_strategy == BKG_FIRST:
+            self.__set(F_BACKGROUND, workspace, background_pixels)
+
+        workspace.object_set.add_objects(objects, self.object_name.value)
+
+        # Make outlines
+        outline_image = centrosome.outline.outline(objects.segmented)
+        if self.should_save_outlines.value:
+            out_img = cpi.Image(outline_image.astype(bool),
+                                parent_image = input_image)
+            workspace.image_set.add(self.save_outlines.value, out_img)
+
+        # Save measurements
+
+        workspace.measurements.add_measurement(self.object_name.value, M_OBJECT_FEATURES_OBJECT_QUALITY,
+                                               objects_qualities)
+
+        cpmi.add_object_location_measurements(workspace.measurements,
+                                              self.object_name.value,
+                                              objects.segmented)
+
+        cpmi.add_object_count_measurements(workspace.measurements,
+                                           self.object_name.value, np.max(objects.segmented))
+
+    def set_params_from_ui(self, params):
+        def update_params(next_name, first_name, random_name, ui_value):
```

Comment:
It uses the `params` and is closely tied with `set_params_from_ui` so I would prefer not to let it loose.